### PR TITLE
[MIG] sale_automatic_workflow: Migration to 14.0

### PR DESCRIPTION
**Pull request by phuctranfxvn**

### CODE REVIEW

**Comment by atchuthan** [comment]
Please check the old PR https://github.com/OCA/sale-workflow/pull/1321, Please review it, and let's merge

**Review by rousseldenis — APPROVED** [review]

**Comment by nilshamerlinck** [comment]
> Please check the old PR #1321, Please review it, and let's merge

Hi @atchuthan oups, seems this previous PR was missed, probably because it was not announced in https://github.com/OCA/sale-workflow/issues/1274 :/

To be pragmatic, given that https://github.com/OCA/sale-workflow/pull/1321 is red and has some fixes pending while this one passes and works, I think this one should be merged

File: sale_automatic_workflow/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                    float_compare(
+                        move.quantity_done,
+                        move.product_qty,
+                        precision_rounding=rounding,
+                    )
+                    == -1
+                ):
+                    for move_line in move.move_line_ids:
+                        move_line.qty_done = move_line.product_uom_qty
+            picking.with_context(skip_overprocessed_check=True).button_validate()
```
Comment (florian-dacosta) [inline sale_automatic_workflow/models/stock_picking.py]: Context `skip_overprocessed_check` does not exist anymore in stock module. I think `skip_immediate` should be used instead

Comment (phuctranfxvn) [reply sale_automatic_workflow/models/stock_picking.py]: Yes, you are right, skip_overprocessed_check is deprecated. fixed in the latest update

**Review by atchuthan — APPROVED** [review]
Code review :+1:

**Comment by atchuthan** [comment]
@florian-dacosta Please check the changes and update the review status

**Comment by grindtildeath** [comment]
@phuctranerp Hello, can you please cherry-pick https://github.com/OCA/sale-workflow/pull/1492 ?
I guess the domain needs to be changed for v14.0 by using new field `posted_before`: https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L1889

**Comment by phuctranfxvn** [comment]
> @phuctranerp Hello, can you please cherry-pick #1492 ?
> I guess the domain needs to be changed for v14.0 by using new field `posted_before`: https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L1889

Hi @grindtildeath

Yes sure, done

Thanks

**Review by atchuthan — APPROVED** [review]

File: sale_automatic_workflow/data/automatic_workflow_data.xml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        <field name="model_id">sale.order</field>
+        <field
+            name="domain"
+        >[('state','in',['sale','done']),('invoice_status','=','to invoice')]</field>
+        <field name="user_id" ref="base.user_root" />
+    </record>
+    <record id="automatic_workflow_validate_invoice_filter" model="ir.filters">
+        <field name="name">Automatic Workflow Validate Invoice Filter</field>
+        <field name="model_id">account.move</field>
+        <field name="domain">[('state', '=', 'draft'), ('name', '=', '/')]</field>
```
Comment (grindtildeath) [inline sale_automatic_workflow/data/automatic_workflow_data.xml]:
```suggestion
        <field name="domain">[('state', '=', 'draft'), ('posted_before', '=', False)]</field>
```

**Review by grindtildeath — COMMENTED** [review]
:wink:

**Comment by rousseldenis** [comment]
@phuctranerp I relaunched Travis as a fix has been merged for sale_order_lot_generator. Could you check pre-commit ?

**Comment by phuctranfxvn** [comment]
> @phuctranerp I relaunched Travis as a fix has been merged for sale_order_lot_generator. Could you check pre-commit ?

I run the precommit and the travis is green now.

**Review by florian-dacosta — APPROVED** [review]

**Pull request merged**